frontend/components/Documents.js

```diff
@@ -68,7 +68,7 @@ const Documents = () => {
             },
             body: JSON.stringify({
                 title: newDocData.title,
-                year: typeof newDocData.year === "string" ? null : newDocData.year,
+                year: newDocData.year,
```

Comment:
As we discussed in person, this is likely the only change needed to resolve the immediate issue. If you can, I'd recommend undoing the other changes in this PR and we can simply merge this one change in. Let me know if you have any questions about how to do that!

Reply:
I just tried to undo the changes manually and hopefully that worked. Would there be an easier way to do that?

Reply:
You could use [`git-revert`](https://git-scm.com/docs/git-revert), but that seems unnecessarily complicated for this kind of change, so I think the manual adjustment is good.